modules/op-using-tekton-chains-to-sign-and-verify-image-and-provenance.adoc

```diff
@@ -0,0 +1,143 @@
+// This module is included in the following assembly:
+//
+// *cicd/pipelines/using-tekton-chains-for-pipelines-supply-chain-security.adoc
+
+:_content-type: PROCEDURE
+[id="using-tekton-chains-to-sign-and-verify-image-and-provenance_{context}"]
+= Using Tekton Chains to sign and verify image and provenance
+
+[role="_abstract"]
+Cluster administrators can use Tekton Chains to sign and verify images and provenances, by performing the following tasks:
+
+* Create an encrypted x509 key pair and save it as a Kubernetes secret.
+* Set up authentication for the OCI registry to store images, image signatures, and signed image attestations.
+* Configure Tekton Chains to generate and sign provenance.
+* Create an image with Kaniko in a task run.
+* Verify the signed image and the signed provenance.
+
+[discrete]
+.Prerequisites
+Ensure that the following are installed on the cluster:
+
+* {pipelines-title} Operator
+* Tekton Chains
+* link:https://docs.sigstore.dev/cosign/installation/[Cosign]
+* link:https://docs.sigstore.dev/rekor/installation/[Rekor]
+* link:https://stedolan.github.io/jq/[jq]
+
+[discrete]
+.Procedure
+
+. Create an encrypted x509 key pair and save it as a Kubernetes secret:
++
+[source,terminal]
+----
+$ cosign generate-key-pair k8s://tekton-chains/signing-secrets
+----
++
+Provide a password when prompted. Cosign stores the resulting private key as part of the `signing-secrets` Kubernetes secret in the `tekton-chains` namespace, and writes the public key to the `cosign.pub` local file.
+
+. Configure authentication for the image registry.
+
+.. To configure the Tekton Chains controller for pushing signature to an OCI registry, use the credentials associated with the service account of the task run. For detailed information, see the "Authenticating to an OCI registry" section.
```

Comment:
We should add a link to that section above, right?

Reply:
It's unfortunate that we aren't allowed to cross-reference one module from another, when they are part of the same assembly. @Preeticp - Any guideline to do what Shubham asked?

Reply:
There has been a [discussion](https://coreos.slack.com/archives/C85JYPHL3/p1649250533446979) about this on Slack, and my approach is acceptable. So for now, it remains the way it's written.